dask/array/utils.py

```diff
@@ -119,6 +119,9 @@ def compute_meta(func, _dtype, *args, **kwargs):
         meta = func(*args_meta)
     else:
         try:
+            # some reduction functions need to know they are computing meta
+            if has_keyword(func, "computing_meta"):
+                kwargs_meta["computing_meta"] = True
```

Comment:
Are there cases where a function with a `computing_meta` parameter would ever be called with `computing_meta=False`? If not, could we get rid of all `if computing_meta:` checks in functions with a `computing_meta=` keyword argument?

Reply:
`computing_meta=False` is the default, we only set it to `True` when we are computing a `meta` reduction. Not sure if this is what you asked, but this is just for those functions to know that the data is actually a `meta` array.

Reply:
I guess my question is why are we handling the setting of `computing_meta` this way (based on the existence of the `computing_meta` keyword) as opposed to specifying `computing_meta=True` when a `meta` reduction is being computed?

For example, in this code block:

https://github.com/dask/dask/blob/842d96b7854371b435092a64812c1a282c0e1d51/dask/array/reductions.py#L166-L168

IIUC the value of the `computing_meta` parameter in like 167 doesn't seem to matter as the behavior of the `compute_meta` function is dictated by whether or not `chunk` has a `computing_meta` keyword, not the actual value of `computing_meta` being passed to the `compute_meta` function.

Reply:
I get it now. So `compute_meta` is a general-purpose function, that should be usable with any array function, thus, it can, for example, be used with `np.sum`, that doesn't have a `computing_meta` argument, that's why we need to test for that first ensuring we only set it for those reduction functions that need to know they are computing `meta`.

Reply:
Gotcha, thanks for the clarification. Then is the `try` branch specifying `computing_meta=True` necessary here?

https://github.com/dask/dask/blob/842d96b7854371b435092a64812c1a282c0e1d51/dask/array/reductions.py#L165-L174

If not, I think it's best to remove it

Reply:
It is necessary. Not all reduction functions take `computing_meta`, the ones that require that are mostly those that would return/take a `dict` instead of an array. The remaining ones maintain the NumPy API (`chunk.sum`, `chunk.min`, etc.).

Reply:
I see, the `try:` is for reduction functions which support `computing_meta` and the `except TypeError` is for all functions that don't (like `chunk.sum`). Thanks for the explanation, not sure why that wasn't clear to me before

Reply:
This part of the code is particularly complicated, it was probably the most difficult one to get right when implementing `_meta`, so I can definitely understand why that isn't clear. :)